src/atopile/front_end.py

```diff
@@ -382,11 +392,22 @@ def survey(
     ) -> Context:
         surveyor = cls()
         context = Context(file_path=file_path, scope_ctx=ctx, refs={})
-        for ref, item in surveyor.visit(ctx):
+        for ref, (item, item_ctx) in surveyor.visit(ctx):
             if ref in context.refs:
-                raise errors.UserKeyError.from_ctx(
-                    ctx, f"Duplicate declaration of {ref}"
-                )
+                ex = errors.UserKeyError.from_ctx(item_ctx, f'"{ref}" already declared')
+                # Downgrade the error if we're just importing
+                # the same thing multiple times
+                first_of_dup = context.refs[ref]
```

Comment:
There's a few cases where, in the std library we have duplicated imports etc...
Ignoring that for now